drivers/tapdisk-nbdserver.c

```diff
@@ -1306,10 +1309,15 @@ tapdisk_nbdserver_clientcb(event_id_t id, char mode, void *data)
 	return;
 
 fail:
+	if (vreq)
+		tapdisk_nbd_server_free_vreq(client, vreq, false);
+
 	tapdisk_nbdserver_free_client(client);
+	return;
+
 free:
 	if (vreq)
```

Comment:
I think this is dead code now after the addition of the `return` on line 1316. The only path to get here is via `TAPDISK_NBD_CMD_DISC` which doesn't ever set `vreq`.

Reply:
Actually, it looks like it was dead in the previous sets of changes. Just cleaning the whole thing up.